Dockerfile

```diff
@@ -1,70 +1,23 @@
- && micromamba clean -ayf \
- && find /opt/conda/ -follow -type f -name '*.a' -delete \
- && find /opt/conda/ -follow -type f -name '*.pyc' -delete \
- && find /opt/conda/ -follow -type f -name '*.js.map' -delete
- ENTRYPOINT [ "micromamba", "run", "-n", "hydromt" ]
- CMD ["hydromt","--models"]
+FROM debian:bookworm-slim as base
+ARG PIXIENV
+WORKDIR /home/hydromt
+RUN apt-get update && apt-get install -y curl
+RUN curl -fsSL https://pixi.sh/install.sh | bash
+ENV PATH=/root/.pixi/bin:$PATH
+COPY pixi.toml pixi.lock pyproject.toml README.rst ./
+COPY data/ ./data
+COPY examples/ ./examples
+COPY hydromt/ ./hydromt
+RUN pixi run -e ${PIXIENV} install-hydromt
+ENV RUNENV="${PIXIENV}"
+RUN echo "pixi run -e ${RUNENV} \$@" > /run_pixi.sh
+ENTRYPOINT ["sh", "/run_pixi.sh"]
+CMD ["hydromt","--models"]
 
-FROM  mambaorg/micromamba:1.4-alpine AS full
-ENV HOME=/home/mambauser
-WORKDIR ${HOME}
-USER mambauser
-COPY full-environment.yml pyproject.toml README.rst ${HOME}/
-RUN micromamba create -f full-environment.yml -y --no-pyc \
- && micromamba clean -ayf \
- && rm -rf ${HOME}/.cache \
- && find /opt/conda/ -follow -type f -name '*.a' -delete \
- && find /opt/conda/ -follow -type f -name '*.pyc' -delete \
- && find /opt/conda/ -follow -type f -name '*.js.map' -delete  \
- && rm full-environment.yml
-COPY data/ ${HOME}/data
-COPY examples/ ${HOME}/examples
-COPY tests/ ${HOME}/tests
-COPY hydromt/ ${HOME}/hydromt
-RUN micromamba run -n hydromt pip install . --no-cache-dir --no-compile --disable-pip-version-check --no-deps\
- && micromamba clean -ayf \
- && find /opt/conda/ -follow -type f -name '*.a' -delete \
- && find /opt/conda/ -follow -type f -name '*.pyc' -delete \
- && find /opt/conda/ -follow -type f -name '*.js.map' -delete
- ENTRYPOINT [ "micromamba", "run", "-n", "hydromt" ]
- CMD ["hydromt","--models"]
+FROM base as min
+COPY tests/ ./tests
 
-FROM  mambaorg/micromamba:1.4-alpine AS slim
-ENV HOME=/home/mambauser
-WORKDIR ${HOME}
-USER mambauser
-COPY slim-environment.yml pyproject.toml README.rst ${HOME}/
-RUN micromamba create -f slim-environment.yml -y --no-pyc \
- && rm -rf ${HOME}/.cache \
- && micromamba clean -ayf \
- && find /opt/conda/ -follow -type f -name '*.a' -delete \
- && find /opt/conda/ -follow -type f -name '*.pyc' -delete \
- && find /opt/conda/ -follow -type f -name '*.js.map' -delete  \
- && rm slim-environment.yml
-COPY data/ ${HOME}/data
-COPY examples/ ${HOME}/examples
-COPY hydromt/ ${HOME}/hydromt
-RUN micromamba run -n hydromt pip install . --no-cache-dir --no-compile --disable-pip-version-check --no-deps\
- && micromamba clean -ayf \
- && find /opt/conda/ -follow -type f -name '*.a' -delete \
- && find /opt/conda/ -follow -type f -name '*.pyc' -delete \
- && find /opt/conda/ -follow -type f -name '*.js.map' -delete
- ENTRYPOINT [ "micromamba", "run", "-n", "hydromt" ]
- CMD ["hydromt","--models"]
+FROM base as full
+COPY tests/ ./tests
+
+FROM base as slim
```

Comment:
For me it's not clear why tests should be in full and minimal, but not in slim. Should the examples be in min?

Reply:
I think that was a mistake of mine. examples and tests should only be in full I think.

Reply:
done